### PR TITLE
TASK: Improve related nodes view for multi-site, dimension and workspaces

### PR DESCRIPTION
**Pull request by aertmann**
Improves the asset related nodes view to show where nodes can actually be
found across workspaces, content dimensions and sites.

Additionally improves the workspaces module changes overview

### CODE REVIEW

**Comment by aertmann** [comment]
Before:
image
After:
image

Before:
image
After:
image

File: TYPO3.Neos/Classes/TYPO3/Neos/Controller/Module/Management/AssetController.php [hunk 1]
```diff
@@ -89,6 +89,18 @@ class AssetController extends \TYPO3\Media\Controller\AssetController
...
 
     /**
+     * @Flow\Inject
+     * @var DomainUserService
+     */
+    protected $domainUserService;
+
+    /**
+     * @var ContentDimensionPresetSourceInterface
+     * @Flow\Inject
```
Comment (johannessteu) [inline TYPO3.Neos/Classes/TYPO3/Neos/Controller/Module/Management/AssetController.php]: All other class properties define first `@Flow\Inject` and then `@var`. Should be snyced

File: TYPO3.Neos/Classes/TYPO3/Neos/Controller/Module/Management/AssetController.php [hunk 2]
```diff
@@ -150,22 +152,38 @@ public function deleteAction(\TYPO3\Media\Domain\Model\Asset $asset)
      * @param Asset $asset
      * @return void
      */
-    public function relatedDocumentNodesAction(Asset $asset)
+    public function relatedNodesAction(Asset $asset)
```
Comment (johannessteu) [inline TYPO3.Neos/Classes/TYPO3/Neos/Controller/Module/Management/AssetController.php]: Method doc should also be updated to `Get Related Nodes from an asset`

File: TYPO3.Neos/Classes/TYPO3/Neos/Service/LinkingService.php [hunk 3]
```diff
@@ -20,11 +20,14 @@
...
 use TYPO3\Media\Domain\Model\AssetInterface;
 use TYPO3\Media\Domain\Repository\AssetRepository;
+use TYPO3\Neos\Domain\Repository\SiteRepository;
 use TYPO3\Neos\Domain\Service\ContentContext;
 use TYPO3\Neos\Domain\Service\NodeShortcutResolver;
+use TYPO3\Neos\Domain\Service\SiteService;
 use TYPO3\Neos\Exception as NeosException;
 use TYPO3\Neos\TYPO3CR\NeosNodeServiceInterface;
 use TYPO3\TYPO3CR\Domain\Model\NodeInterface;
+use TYPO3\TYPO3CR\Domain\Utility\NodePaths;
```
Comment (johannessteu) [inline TYPO3.Neos/Classes/TYPO3/Neos/Service/LinkingService.php]: Are those classes used at all in the linking service?

File: TYPO3.Neos/Resources/Private/Partials/Module/Management/Workspaces/ContentChangeDiff.html [hunk 4]
```diff
@@ -3,19 +3,27 @@
...
                 </f:if>
                 {change.node.label}
             </th>
+            <f:if condition="{contentDimensions}">
+                <th>
+                    <f:render partial="Module/Shared/NodeContentDimensionsInformation.html" arguments="{contentDimensions: contentDimensions, node: change.node.nodeData}" />
+                </th>
+            </f:if>
+            <th>
+                <span title="{neos:backend.translate(id: 'lastModification')}: {change.node.lastModificationDateTime -> f:format.date(format: 'Y-m-d H:i')}" data-neos-toggle="tooltip">{change.node.lastModificationDateTime -> m:format.relativeDate()}</span>
```
Comment (johannessteu) [inline TYPO3.Neos/Resources/Private/Partials/Module/Management/Workspaces/ContentChangeDiff.html]: We should show here at least a label what kind of time is shown here. As an editor i'm not sure if it is the time when published, modified, created or whatever 
screen shot 2016-04-08 at 09 07 49

**Comment by johannessteu** [comment]
Hey @aertmann , looks really good! Just left some nitpicks

**Comment by johannessteu** [comment]
There should also be some margin between the icon and the label 
screen shot 2016-04-08 at 09 00 53

File: TYPO3.Neos/Classes/TYPO3/Neos/Service/LinkingService.php (hunk 3 above)
Comment (aertmann) [inline TYPO3.Neos/Classes/TYPO3/Neos/Service/LinkingService.php]: whoops, sneaked in from another change

**Comment by aertmann** [comment]
@johannessteu thanks for the feedback, fixed the comments

regarding the date, there is already
image

**Comment by johannessteu** [comment]
Yeah, i saw that tooltip, but is that enough!? There is no real indication for the editor like "Hey, here is a tooltip" and you maybe just get the information if you're lucky and hover the date there. Maybe a small Clock icon or something next to the date?

**Comment by aertmann** [comment]
I'd say it is, if you're in doubt you take a closer look otherwise you ignore it. Tried with labels first and found it unnecessary and cluttering, so removed them again. I like the clock idea though, will add that.

**Comment by aertmann** [comment]
Added icon as suggested 
image

**Comment by johannessteu** [comment]
👍

**Comment by radmiraal** [comment]
Made a PR to askes branch to make the usage of the UserService a bit cleaner, would be fine with merging this when that PR is merged

**Comment by aertmann** [comment]
🎉 thanks for the reviews and pr @radmiraal

**Pull request merged**